fix builds for macos 12.6

markcmiller86 opened the pull request:
### Description

Resolves # <!-- If this PR is unrelated to a ticket, please erase this line -->

<!-- Please include a summary of the change and any other information and instructions to help reviewers understand the work -->

<!-- Note that all the checklist items in various bulleted lists below end with ~~ characters.
     This is to facilitate striking them out when they do not apply.
     One just has to add ~~ characters just before the opening square bracket [ -->

### Type of change

<!-- Please check one of the boxes below -->

* [ ] Bug fix~~
* [ ] New feature~~
* [ ] Documentation update~~
* [ ] Other~~ <!-- please explain with a note below -->

### How Has This Been Tested?

<!-- Please describe the tests you've added or any tests that already cover this change. Include relevant information, such as which operating system you tested on. -->

### Reminders:

- Please follow the [style guidelines][1] of this project.
- Please perform a self-review of your code before submitting a PR and asking others to review it.
- Please assign reviewers (see [VisIt's PR procedures][2] for more information).

### Checklist:

<!-- For items in this checklist that do not apply, simply insert two tilde chars, `~~`, just ahead of the left bracket char, `[` at the beginning of a line. Each line ends with two tilde chars to make doing such ~~strikeouts~~ easy. -->

- [ ] I have commented my code where applicable.~~
- [ ] I have updated the release notes.~~
- [ ] I have made corresponding changes to the documentation.~~
- [ ] I have added debugging support to my changes.~~
- [ ] I have added tests that prove my fix is effective or that my feature works.~~
- [ ] I have confirmed new and existing unit tests pass locally with my changes.~~
- [ ] I have added new baselines for any new tests to the repo.~~
- [ ] I have NOT made any changes to [*protocol* or *public interfaces*][3] in an RC branch.~~

[1]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/StyleGuide.html
[2]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/pr_create.html#reviewers
[3]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/RCDevelopment.html#communication-protocols-and-public-apis

markcmiller86 committed:
fix builds for macos 12.6

markcmiller86 committed:
use lzma compression for packaging

markcmiller86 committed:
remove unneeded arg to tar -x

biagas commented on src/tools/dev/scripts/bv_support/bv_hdf5.sh:
```diff
@@ -489,6 +489,35 @@ EOF
 
+function apply_hdf5_resize_configs_are_equal_patch
+{
+    info "Patching hdf5 1.8.14 for isatty"
```
Is this comment correct?

markcmiller86 replied:
Doh! I don't think so. Will fix shortly.

markcmiller86 replied:
```suggestion
    info "Patching hdf5 1.8.14 for resize configs are equal"
```

markcmiller86 commented on src/CMakeLists.txt:
```diff
@@ -2346,7 +2346,7 @@ IF (NOT WIN32)
         SET (CPACK_BUNDLE_STARTUP_COMMAND "${VISIT_BINARY_DIR}/exe/visit_macos_launcher")
     ELSE(APPLE AND VISIT_CREATE_APPBUNDLE_PACKAGE)
-        SET(CPACK_GENERATOR "TGZ")
+        SET(CPACK_GENERATOR "TXZ")
```
This change leads to 100Mb smaller tarball on mac. Would assume similar savings on other platforms. If we could control flags to xz, we could improve that a bit more. We could do that if we used Python tools to tar and compress...as we do in the top level `data` dir.

markcmiller86 commented on src/tools/dev/masonry/masonry.py:
```diff
@@ -412,7 +412,9 @@ def execute(self,base,key,tag,parent_res):
-            cmd = "hdiutil create -srcFolder %s -o %s" % (src_folder, temp_dmg)
+            # ULMO format (xz) is 1/2 the size of default format (UDZO which is zlib level 1)
+            #cmd = "hdiutil create -srcFolder %s -o %s" % (src_folder, temp_dmg)
+            cmd = "hdiutil create -format ULMO -srcFolder %s -o %s" % (src_folder, temp_dmg)
```
This change leads to 275Mb smaller `.dmg` file on mac

brugger1 commented on src/tools/dev/scripts/bv_support/bv_hdf5.sh:
```diff
@@ -489,6 +489,35 @@ EOF
+                           hbool_t compare_init,
+EOF
+    if [[ $? != 0 ]] ; then
+        warn "HDF5 1.8.14 resize_configs_are_equal patch."
```
Should the warning indicate that the patch failed?

markcmiller86 replied:
Yes, it should. I'll fix shortly

markcmiller86 replied:
```suggestion
        warn "HDF5 1.8.14 resize_configs_are_equal patch failed."
```

markcmiller86 committed:
Update src/tools/dev/scripts/bv_support/bv_hdf5.sh

markcmiller86 committed:
Update src/tools/dev/scripts/bv_support/bv_hdf5.sh

markcmiller86 committed:
use updated libtoolized mpich on macOS

markcmiller86 committed:
Merge branch 'rel-mcm86-11apr23-build-fixes' of github.com:visit-dav/visit into rel-mcm86-11apr23-build-fixes

markcmiller86 commented on src/tools/dev/scripts/bv_support/bv_mpich.sh:
```diff
@@ -22,13 +22,23 @@ function bv_mpich_depends_on
-    export MPICH_URL=${MPICH_URL:-http://www.mpich.org/static/tarballs/${MPICH_VERSION}}
-    export MPICH_MD5_CHECKSUM="9ed4cabd3fb86525427454381b25f6af"
-    export MPICH_SHA256_CHECKSUM="fe551ef29c8eea8978f679484441ed8bb1d943f6ad25b63c235d4b9243d551e5"
+    if [[ "$OPSYS" == "Darwin" ]]; then
```
Updating mpich with newer libtool involves quite a bit. No easy way to just patch it. So, I created a new `third-party` asset named `mpich-3.3.1-libtool-2.4.6` which expands also to `mpich-3.3.1` (honestly should be harmless if its used on non-macOS systems too).

markcmiller86 replied:
This commit is needed for macOS 12.6. So I cherry picked it to #19293

markcmiller86 commented on src/tools/dev/scripts/bv_support/bv_mpich.sh:
```diff
@@ -22,13 +22,23 @@ function bv_mpich_depends_on
-    export MPICH_URL=${MPICH_URL:-http://www.mpich.org/static/tarballs/${MPICH_VERSION}}
-    export MPICH_MD5_CHECKSUM="9ed4cabd3fb86525427454381b25f6af"
-    export MPICH_SHA256_CHECKSUM="fe551ef29c8eea8978f679484441ed8bb1d943f6ad25b63c235d4b9243d551e5"
+    if [[ "$OPSYS" == "Darwin" ]]; then
```
Updating mpich with newer libtool involves quite a bit. No easy way to just patch it. So, I created a new `third-party` asset named `mpich-3.3.1-libtool-2.4.6` which expands also to `mpich-3.3.1` (honestly should be harmless if its used on non-macOS systems too).

markcmiller86 committed:
patches for macOS 12.6

markcmiller86 committed:
fix builds for macOS 12

markcmiller86 committed:
fix configure for all OSs

markcmiller86 commented on src/tools/dev/scripts/bv_support/bv_advio.sh:
```diff
@@ -185,6 +185,12 @@ function build_advio
+    else
+        sed -i.orig -e 's/^main()/int main()/' configure
+    fi
     info "Invoking command to configure AdvIO"
```
Many try-run blocks for sizes of types in configure defined a `main()` with no type which causes modern compilers to fail outright. This `sed` command fixes them. Its a little different on macOS then other *nix systems. Windows?

biagas replied:
@markcmiller86 Sorry, didn't see the question about Windows sooner.  Not using the build_visit script on Windows, so not an issue.

markcmiller86 commented:
@biagas and @brugger1 I've pulled this out of **DRAFT** mode. I think it is about ready.

markcmiller86 committed:
more build fixes

markcmiller86 committed:
fix typo

markcmiller86 commented:
Ok, so my attempt to build from a fresh state for this branch did fail in Qt. And I need to investigate and fix.

markcmiller86 commented:
The problem with Qt5 on macOS is that it does not support python3. It requires a python2 (for at least QtWebEngine anyways...maybe we should explicitly disable that?). On my macOS 12.6.7, with a *vanilla* `PATH`, I have a `python3` but no `python` or `python2` and that is what Qt is looking for. Now, I have a version 2 python there in `/opt/local/bin`. But, there is *a ton of other stuff* in `/opt/local/bin`. So, I don't want `/opt/local/bin` in my path. So, I added a symlink named `python` in a random dir and added that dir to my path so hopefully it will find that version 2 python

markcmiller86 commented:
I wound up having to add `-skip webengine` to Qt configure.

brugger1 commented:
There is a patch in bv_qt.sh that should handle the python issue, but it is only applied for Qt 5.10.1. It looks like it should also be added for Qt 5.14.2.

https://github.com/visit-dav/visit/blob/02e4b02b63c98bd5780bc7709713836a71a70ede/src/tools/dev/scripts/bv_support/bv_qt.sh#L287-L296

markcmiller86 committed:
snapshot

cyrush commented:
PR #18848 will include the changes for macOS.

cyrush commented:
closing in lieu of PR #18848 and #19293
